File: problemas/iniciante/1435/1435.py
def main():
    while True:
        N = int(input())
        if N == 0:
            break

        saida = ''
        for i in range(N):
            saida += f"{1:3d}"
            for j in range(1, N):
                saida += f"{(min(min(i, N - i - 1), min(j, N - j - 1)) +1):4d}"
            
            saida += "\n"
        
        print(saida)

if __name__ == "__main__":
    main()

File: problemas/iniciante/1435/1435.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = "";
let lineCount = 0;

process.stdin.on('data', (chunk) => {
    input += chunk;
});

process.stdin.on('end', () => {
    let entrada = input.split('\n');

    while (entrada.length) {
        let N = parseInt(entrada.shift().trim());

        if (N === 0) {
            break;
        }

        let saida = '';

        for (let i = 0; i < N; ++i) {
            saida += (1).toString().padStart(3);

            for (let j = 1; j < N; ++j) {
                saida += (Math.min(Math.min(i, N - i - 1), Math.min(j, N - j - 1)) + 1).toString().padStart(4);
            }
            saida += '\n';
        }
        console.log(saida);
    }

});